### PR TITLE
Add path filter to CI

### PR DESCRIPTION
**Pull request by boraarslan**

### CODE REVIEW

**Comment by boraarslan** [comment]
Since we need for the job to run, I only filtered the cargo command step.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -66,6 +63,14 @@ jobs:
           --health-retries 5
     steps:
       - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
+            src:
```
Comment (guilload) [inline .github/workflows/ci.yml:70]:
```suggestion
            qw-src:
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -66,6 +63,14 @@ jobs:
...
+              - quickwit/**
+            ui:
```
Comment (guilload) [inline .github/workflows/ci.yml:72]:
```suggestion
            ui-src:
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -99,5 +104,6 @@ jobs:
         if: matrix.task.name == 'cargo nextest'
         uses: taiki-e/install-action@nextest
       - name: ${{ matrix.task.name }}
+        if: steps.changes.outputs.src == 'true' && steps.changes.outputs.ui == 'false'
```
Comment (guilload) [inline .github/workflows/ci.yml:107]:
```suggestion
        if: steps.modified-files.outputs.qw-src == 'true' && steps.changes.outputs.ui-src == 'false'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -66,6 +63,14 @@ jobs:
           --health-retries 5
     steps:
       - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: changes
```
Comment (guilload) [inline .github/workflows/ci.yml:67]:
```suggestion
        id: modified-files
```

**Review by guilload — APPROVED** [review]

**Pull request merged**